app/medInria/medRegistrationWorkspace.cpp

```diff
@@ -43,23 +42,13 @@ class medRegistrationWorkspacePrivate
     medLayerParameterGroup *movingLayerGroup;
 };
 
-medRegistrationWorkspace::medRegistrationWorkspace(QWidget *parent) : medAbstractWorkspace(parent), d(new medRegistrationWorkspacePrivate)
+medRegistrationWorkspace::medRegistrationWorkspace(QWidget *parent) : medSelectorWorkspace(parent, staticName(), new medRegistrationSelectorToolBox(parent, staticName())), d(new medRegistrationWorkspacePrivate)
 {
-    d->selectorToolBox = new medRegistrationSelectorToolBox(parent);
-    d->selectorToolBox->setWorkspace(this);
-    this->addToolBox(d->selectorToolBox);
-    d->selectorToolBox->setTitle(this->name()); // get workspace name
-
     connect(this->stackedViewContainers(), SIGNAL(currentChanged(int)), this, SLOT(updateUserLayerClosable(int)));
-    connect(d->selectorToolBox, SIGNAL(movingDataRegistered(medAbstractData*)), this, SLOT(updateFromRegistrationSuccess(medAbstractData*)));
-    connect(d->selectorToolBox, SIGNAL(destroyed()), this, SLOT(removeSelectorInternToolBox()));
+    connect(selectorToolBox(), SIGNAL(movingDataRegistered(medAbstractData*)), this, SLOT(updateFromRegistrationSuccess(medAbstractData*)));
+    connect(selectorToolBox(), SIGNAL(destroyed()), this, SLOT(removeSelectorInternToolBox()));
```

Comment:
this is not needed, as said below. selectorToolBox() is already removed in its class